pkg/cmd/cli/cmd/deploy.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/cli/describe"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+const newCmdDeployDescription = `
+View, start and restart deployments.
+
+If no options are given, view the latest deployment.
+`
+
+func NewCmdDeploy(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	var forceDeploy bool
+	var retryDeploy bool
+
+	cmd := &cobra.Command{
+		Use:   "deploy <deploymentConfig>",
+		Short: "View, start and restart deployments.",
+		Long:  newCmdDeployDescription,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 || len(args[0]) == 0 {
+				fmt.Println(cmdutil.UsageError(cmd, "A deploymentConfig name is required."))
+				return
+			}
+			if forceDeploy && retryDeploy {
+				fmt.Println(cmdutil.UsageError(cmd, "Only one of --force or --retry is allowed."))
+				return
+			}
+
+			configName := args[0]
+
+			osClient, kubeClient, err := f.Clients()
+			cmdutil.CheckErr(err)
+
+			namespace, err := f.DefaultNamespace()
+			cmdutil.CheckErr(err)
+
+			config, err := osClient.DeploymentConfigs(namespace).Get(configName)
+			cmdutil.CheckErr(err)
+
+			switch {
+			case forceDeploy:
+				err = runForceDeployment(config, kubeClient, osClient, out)
+			case retryDeploy:
+				err = runRetryDeployment(config, kubeClient, out)
+			default:
+				d := describe.NewLatestDeploymentDescriber(osClient, kubeClient)
+				s, err := d.Describe(config.Namespace, config.Name)
+				cmdutil.CheckErr(err)
+				fmt.Printf(s)
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().BoolVar(&forceDeploy, "force", false, "Start a new deployment now.")
+	cmd.Flags().BoolVar(&retryDeploy, "retry", false, "Retry the latest failed deployment.")
+
+	return cmd
+}
+
+func runForceDeployment(config *deployapi.DeploymentConfig, kubeClient kclient.Interface, osClient client.Interface, out io.Writer) error {
```

Comment:
Add a command test that mocks some of it (see what I've done in https://github.com/smarterclayton/origin/commit/b27db760b83bd6466ce258f41f1c8049da51e169 for start-build).  Also add a hack/test-cmd.  Then you can be merged.